apiserver/facades/client/machinemanager/machinemanager.go

```diff
@@ -356,7 +359,7 @@ func (mm *MachineManagerAPI) ProvisioningScript(ctx context.Context, args params
 		return result, errors.Trace(err)
 	}
 
-	icfg, err := InstanceConfig(ctx, mm.controllerConfigService, st, mm.st, mm.cloudService, mm.credentialService, args.MachineId, args.Nonce, args.DataDir)
+	icfg, err := InstanceConfig(ctx, mm.controllerConfigService, st, mm.st, mm.cloudService, mm.credentialService, mm.controllerStore, args.MachineId, args.Nonce, args.DataDir)
```

Comment:
For cases like this, I would be tempted to create an encapsulating service, embed all the constituent services, and pass it as a single argument.